src/main/scala/com/sumologic/sumobot/plugins/BotPlugin.scala

```diff
@@ -127,7 +127,7 @@ abstract class BotPlugin
     def http(request: HttpUriRequest)(func: (IncomingMessage, HttpResponse) => OutgoingMessage): Unit = {
       respondInFuture {
         incoming =>
-          val client = new DefaultHttpClient()
+          val client = HttpClientBuilder.create().build()
```

Comment:
Unfortunately the default client in `apache-http` has a problem. It hangs forever (and blocks the thread) if the connection hangs for some reason.
I suggest to add:
```
    val requestConfig: RequestConfig = RequestConfig.custom
      .setConnectionRequestTimeout(1.minute.toMillis.toInt)
      .setConnectTimeout(1.minute.toMillis.toInt)
      .build
    HttpClientBuilder.create()
      .setDefaultRequestConfig(requestConfig)
      .build()
```
here and everywhere a new client is created (probably extract to place).

The problem is not only theoretical. It has proved to cause other bots to hang before.